authorization-server/src/modules/auth/guards/valid-refresh-token.guard.ts

```diff
@@ -1,15 +1,37 @@
 import { CanActivate, ExecutionContext, Inject } from '@nestjs/common';
 import { AuthService } from '../auth.service';
+import { ConfigService } from '@nestjs/config';
 
 export class ValidRefreshTokenGuard implements CanActivate {
-  constructor(@Inject(AuthService) private readonly authService: AuthService) {}
+  constructor(
+    @Inject(AuthService) private readonly authService: AuthService,
+    @Inject(ConfigService) private readonly config: ConfigService,
+  ) {}
 
   async canActivate(context: ExecutionContext): Promise<boolean> {
-    const { body } = context.switchToHttp().getRequest();
-    if (!body?.refreshToken) {
+    const { body, cookies } = context.switchToHttp().getRequest();
+
+    let refreshToken: string;
+
+    if (
+      this.config.get<boolean>('AUTH_COOKIE_ENABLED') &&
+      this.config.get<boolean>('AUTH_COOKIE_ONLY')
+    ) {
+      refreshToken = (cookies || {})[
+        this.config.get('AUTH_COOKIE_NAME_REFRESH_TOKEN')
+      ];
+    } else if (!this.config.get<boolean>('AUTH_COOKIE_ENABLED')) {
+      refreshToken = body?.refreshToken;
+    } else {
+      refreshToken =
+        body?.refreshToken ||
+        (cookies || {})[this.config.get('AUTH_COOKIE_NAME_REFRESH_TOKEN')];
+    }
```

Comment:
I think this logic can be simplified as shown:
```suggestion
    if (!this.config.get<boolean>('AUTH_COOKIE_ONLY')) {
      refreshToken = body?.refreshToken;
    } else if (this.config.get<boolean>('AUTH_COOKIE_ENABLED')) {
      refreshToken = (cookies || {})[
        this.config.get('AUTH_COOKIE_NAME_REFRESH_TOKEN')
      ];
    }
```

Reply:
it needs to be like this when having the option of tokens in both cookies and bodies